File: doc/src/snippets/qx11embedcontainer/main.cpp
/****************************************************************************
**
** Copyright (C) 2006-$THISYEAR$ $TROLLTECH$. All rights reserved.
**
** This file is part of the $MODULE$ of the Qt Toolkit.
**
** $TROLLTECH_DUAL_LICENSE$
**
** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
**
****************************************************************************/

#include <QtGui>
#include <QX11EmbedContainer>

//! [0]
int main(int argc, char *argv[])
{
    QApplication app(argc, argv);

    if (app.arguments().count() != 2) {
        qFatal("Error - expected executable path as argument");
        return 1;
    }

    QX11EmbedContainer container;
    container.show();

    QProcess process(&container);
    QString executable(app.arguments()[1]);
    QStringList arguments;
    arguments << QString::number(container.winId());
    process.start(executable, arguments);

    int status = app.exec();
    process.close();
    return status;
}
//! [0]

File: doc/src/snippets/qx11embedcontainer/main.java
import com.trolltech.qt.*;
import com.trolltech.qt.core.*;
import com.trolltech.qt.gui.*;

import java.util.*;

public class main
{

//! [0]

This Code is not applicable to Qt Jambi.
public static void main(String args[])
{
    QApplication.initialize(args);

    if (args.length != 2) {
        System.err.println("Error - expected executable path as argument");
        System.exit(1);
    }

    QX11EmbedContainer container = new QX11EmbedContainer();
    container.show();

    QProcess process = new QProcess(container);
    String executable = QApplication.arguments().get(1);
    List<String> arguments = new Vector<String>();
    arguments.add(String.valueOf(container.winId()));
    process.start(executable, arguments);

    QApplication.execStatic();
    QApplication.shutdown();

    process.close();
}
//! [0]

}
